ibis/backends/clickhouse/compiler/relations.py

```diff
@@ -17,79 +16,93 @@ def translate_rel(op: ops.TableNode, **_):
+
+@translate_rel.register
+def _selection(
+    op: ops.Selection, *, table, selections, predicates, sort_keys, aliases, **_
+):
     # needs_alias should never be true here in explicitly, but it may get
     # passed via a (recursive) call to translate_val
-    assert not needs_alias, "needs_alias is True"
-    if needs_alias := isinstance(op.table, ops.Join) and not isinstance(
+    if isinstance(op.table, ops.Join) and not isinstance(
         op.table, (ops.LeftSemiJoin, ops.LeftAntiJoin)
     ):
         args = table.this.args
         from_ = args["from"]
         (join,) = args["joins"]
     else:
         from_ = join = None
-    tr_val = partial(translate_val, needs_alias=needs_alias, **kw)
-    selections = tuple(map(tr_val, op.selections)) or "*"
-    sel = sg.select(*selections, dialect="clickhouse").from_(
-        from_ if from_ is not None else table, dialect="clickhouse"
-    )
+
+    selections = tuple(
+        replace_tables_with_star_selection(node) for node in selections
+    ) or (STAR,)
+
+    # if we're only reprojecting table, return the table
+    if (
+        len(selections) == 1
+        and from_ is None
+        and join is None
+        and not predicates
+        and not sort_keys
+        and all(
+            isinstance(node, sg.exp.Column) and node.this == STAR for node in selections
+        )
+    ):
+        return table
+    sel = sg.select(*selections).from_(from_ if from_ is not None else table)
 
     if join is not None:
         sel = sel.join(join)
 
-    if predicates := op.predicates:
+    if predicates:
         if join is not None:
-            sel = sg.select("*").from_(sel.subquery(kw["aliases"][op.table]))
-        res = functools.reduce(
-            lambda left, right: left.and_(right),
-            (
-                sg.condition(tr_val(predicate), dialect="clickhouse")
-                for predicate in predicates
-            ),
-        )
-        sel = sel.where(res, dialect="clickhouse")
+            sel = sg.select(STAR).from_(sel.subquery(aliases[op.table]))
+        sel = sel.where(*predicates)
 
-    if sort_keys := op.sort_keys:
-        sel = sel.order_by(*map(tr_val, sort_keys), dialect="clickhouse")
+    if sort_keys:
+        sel = sel.order_by(*sort_keys)
 
     return sel
 
 
 @translate_rel.register(ops.Aggregation)
-def _aggregation(op: ops.Aggregation, *, table, **kw):
-    tr_val = partial(translate_val, **kw)
-    tr_val_no_alias = partial(translate_val, render_aliases=False, **kw)
-
-    by = tuple(map(tr_val, op.by))
-    metrics = tuple(map(tr_val, op.metrics))
-    selections = (by + metrics) or "*"
+def _aggregation(
+    op: ops.Aggregation, *, table, metrics, by, having, predicates, sort_keys, **_
+):
+    selections = (by + metrics) or STAR
     sel = sg.select(*selections).from_(table)
 
-    if group_keys := op.by:
-        sel = sel.group_by(*map(tr_val_no_alias, group_keys), dialect="clickhouse")
+    if by:
+        sel = sel.group_by(*range(1, len(by) + 1))
```

Comment:
I think this should be reverted back to the original. The group by indexes are a newish feature and some of our users need the expression compiled in.

Reply:
Fixed.